libs/components/packages/src/schematics/migrations/update-7/ag-grid-28.schematic.ts

```diff
@@ -0,0 +1,195 @@
+        type,
+        version,
+      });
+    }
+    if (agGridAllModulesEnt) {
+      removePackageJsonDependency(tree, `${ENT_MODULE}all-modules`);
+    }
+
+    // Install the new packages.
+    if (
+      needsPackageUpdated ||
+      needsEntPackageUpdated ||
+      agGridAllModules ||
+      agGridAllModulesEnt
+    ) {
+      context.addTask(new NodePackageInstallTask());
+    }
+
+    const ignoreList = tree.exists('.gitignore')
+      ? tree.readText('.gitignore')
+      : '';
+    const ig = ignore().add(ignoreList);
+
+    tree.visit((filePath) => {
+      // If the file is not a TypeScript file or if it is ignored by git, we can skip it.
+      if (!filePath.endsWith('.ts') || ig.ignores(relative('/', filePath))) {
+        return;
+      }
+      const content = tree.readText(filePath);
+      let updatedContent = content;
+
+      // Prompt the user to moderate the use of AG Grid modules
+      if (
+        updatedContent.includes(ANY_MODULE) ||
+        updatedContent.includes(ENT_MODULE)
+      ) {
+        warnOnce(`\n
+        AG Grid recommends not mixing module and package imports.
+        https://ag-grid.com/angular-data-grid/modules/\n\n`);
+      }
+      // Replace `@ag-grid-community/all-modules` with `ag-grid-community`
+      if (
+        agGridAllModules &&
+        updatedContent.includes(`${ANY_MODULE}all-modules`)
+      ) {
+        updatedContent = updatedContent.replace(
+          /@ag-grid-community\/all-modules/g,
+          AG_GRID
+        );
+      }
+      if (
+        agGridAllModulesEnt &&
+        updatedContent.includes(`${ENT_MODULE}all-modules`)
+      ) {
+        updatedContent = updatedContent.replace(
+          /@ag-grid-enterprise\/all-modules/g,
+          AG_GRID_ENT
+        );
+      }
+      // Replace `@ag-grid-community/angular` with `ag-grid-angular`
+      if (agGridAllModules && updatedContent.includes(`${ANY_MODULE}angular`)) {
+        updatedContent = updatedContent.replace(
+          /@ag-grid-community\/angular/g,
+          'ag-grid-angular'
+        );
+      }
+
+      // Remove `AgGridModule.withComponents` in module import
+      if (
+        (filePath.endsWith('module.ts') || filePath.endsWith('spec.ts')) &&
+        updatedContent.includes(withComponents)
+      ) {
+        let pos: number;
+        while ((pos = updatedContent.indexOf(withComponents)) !== -1) {
+          // Find closing parenthesis after `AgGridModule.withComponents`
+          const end = updatedContent.indexOf(')', pos);
+          // Leave `AgGridModule`, drop `.withComponents`.
+          updatedContent = [
+            updatedContent.substring(0, pos + 12),
+            updatedContent.substring(end + 1),
+          ].join('');
+        }
+      }
+
+      // Column API renamed getSecondaryColumns / setSecondaryColumns
+      if (updatedContent.includes('etSecondaryColumns(')) {
+        updatedContent = updatedContent.replace(
+          /\.(get|set)SecondaryColumns\(/g,
```

Comment:
Just to be safe, it might be good to check if this file also imports from `ag-grid-*` before making the replacement (just in case the file includes a function with the same name but isn't referencing AgGrid).